src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -265,17 +299,17 @@ internal static class UriHelper
         // - It is a RARE case when Unescape actually needs escaping some characters mentioned above.
         //   For this reason it returns a char[] that is usually the same ref as the input "dest" value.
         //
-        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+        internal static unsafe ValueStringBuilder UnescapeString(string input, int start, int end, ref ValueStringBuilder dest,
```

Comment:
We should not be returning ValueStringBuilder.  That makes a copy and is dangerous.